### PR TITLE
feat(twap): add recipient to summary in cancel modal

### LINKED ISSUE #2887: [1.42]No recipient in the Cancelled/Expired pop-up for TWAP order
Reported in https://github.com/cowprotocol/cowswap/pull/2877#issuecomment-1634303780

1. Open TWAP orders page
2. Place a TWAP order to a recipient
3. Once an order is signed and executed, cancel an order or wait till it is expired
4. Check the pop-up

**AR**: there is no recipient in these pop-ups

and no recipient in the cancellation modal: 
image

**ER**: recipient should be present

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

Fixes #2887

This adds a recipient to TWAP orders cancelation modal and popups.

Cancel confirmation modal
Screenshot 2023-07-18 at 14 02 09

# To test
- I've tested it in cancel confirmation modal but not in the popups since I couldn't get them.
- Test in cancel confirm modal and in popups

### CODE REVIEW

File: src/legacy/state/orders/updaters/utils.ts [hunk 1]
```diff
@@ -49,6 +59,10 @@ export function computeOrderSummary({
       // We only have the API order info, let's at least use that
       summary = `Swap ${sellToken} for ${buyToken}`
     }
+
+    if (owner && receiver && receiver !== owner) {
+      summary += ` to ${shortenAddress(receiver)}`
```
Comment (alfetopito) [inline src/legacy/state/orders/updaters/utils.ts:64]: Does this change anything with regular SWAP/LIMIT orders?

Comment (nenadV91) [reply src/legacy/state/orders/updaters/utils.ts:64]: I don't think so

Comment (nenadV91) [reply src/legacy/state/orders/updaters/utils.ts:64]: I don't think so
Tested limit order with recipient and looked good to me

**Review by elena-zh — APPROVED** [review]
I have created #2913 task for the Expired pop-up

**Review by shoom3301 — APPROVED** [review]

**Pull request merged**